src/borg/archiver.py

```diff
@@ -1024,7 +1028,11 @@ def do_prune(self, args, repository, manifest, key):
                     else:
                         if args.output_list:
                             list_logger.info('Pruning archive: %s' % format_archive(archive))
-                        Archive(repository, key, manifest, archive.name, cache).delete(stats, forced=args.forced)
+                        Archive(repository, key, manifest, archive.name, cache,
+                                progress=args.progress).delete(stats, forced=args.forced)
+                        if args.progress:
+                            archives_deleted += 1
+                            logger.info('(%d/%d) archives pruned' % (archives_deleted, to_delete_len))
```

Comment:
... and this ^^^.

So, the only difference is that one is output before and the other after deletion of one archive - and the counter / total.

I think one log line is enough for that (and it maybe should be before deleting the archive).

Check how 1.0-maint handles this and also how it is handled at other places (--list vs. --progress).

Also, the progress output happens on the borg.output.progress logger elsewhere.